examples/bzlmod/MODULE.bazel

```diff
@@ -18,6 +18,7 @@ python.toolchain(
 )
 use_repo(python, "python3_9")
 use_repo(python, "python3_9_toolchains")
+use_repo(python, "python3_9_host_interpreter")
```

Comment:
Chris's PR has been merged. To get the toolchain interpreter as a repo-available interpreter, do:

```
interpreter = use_extension("@rules_python//python/extensions:interpreter.bzl", "interpreter")
interpreter.install(
  name = "python3_9_interpreter",
  python_name = "python3_9",  # the name from the toolchain() call
)
use_repo(interpreter, "python3_9_interpreter")

pip.parse(..., python_interpreter_target = "@python3_9_interpreter//:python")
```